backend/src/main/java/com/cgi/retrospecto/backend/poker/controller/impl/PokerControllerImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.cgi.retrospecto.backend.poker.controller.impl;
+
+import com.cgi.retrospecto.backend.poker.api.PokerConstants;
+import com.cgi.retrospecto.backend.poker.controller.PokerController;
+import com.cgi.retrospecto.backend.poker.controller.dto.*;
+import com.cgi.retrospecto.backend.poker.domain.Room;
+import com.cgi.retrospecto.backend.poker.domain.Story;
+import com.cgi.retrospecto.backend.poker.domain.Vote;
+import com.cgi.retrospecto.backend.poker.exception.RoomNotFoundException;
+import com.cgi.retrospecto.backend.poker.exception.StoryNotFoundException;
+import com.cgi.retrospecto.backend.poker.helper.converter.RoomConverter;
+import com.cgi.retrospecto.backend.poker.helper.converter.StoryConverter;
+import com.cgi.retrospecto.backend.poker.helper.converter.VoteConverter;
+import com.cgi.retrospecto.backend.poker.service.RoomService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.messaging.handler.annotation.DestinationVariable;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.web.bind.annotation.*;
+
+import static com.cgi.retrospecto.backend.poker.api.PokerConstants.ApiConstants.POKER;
+
+@RestController
+@RequestMapping(POKER)
+@CrossOrigin
+public class PokerControllerImpl implements PokerController {
+    @Autowired
+    private RoomService roomService;
+
+    // @formatter:off
+    @Override
+    public ResponseEntity<RoomDTO> createRoom(@RequestBody CreateRoomDTO dto) {
```

Comment:
It is like this because the lack of my JS knowledge. So in frontend I could copy-paste working requests with some modifications. Later I will try to modify the frontend and simplify so these DTOs will be gone.

Reply:
I am not sure I understand completely what you mean, but let me give you an example here. So in the frontend currently you have to provide an object of CreateRoomDTO when you post to this endpoint. There are two easier ways that come to my mind, you can check createBoard() method in BoardPage.vue:

axios.post(`https://www.retrospecto.cloud/board/create/` + this.author)
          .then(response => {

Or go with this if you don't want to use a parameter:

create() {
  // Simple POST request with a JSON body using axios
  const author = "author"
  axios.post("https://test.net/api/poker", author )
    .then(response => this.room = response.data.room)
}

Reply:
I modified the backend to this:

```
public ResponseEntity<RoomDetails> createRoom(@RequestBody String author) {
   log("Creating room: " + author);
}
```

Make the request like this: 

```
axios.post(this.backendUrls.createRoom(), "author").then(...)
```

The log result will be: `Creating room: author=`. So it adds '=' at the end of it. This is why I had the DTO with a single property. Do you know any fix for this? (other than .substring the String it on the backend?)

Reply:
Nevermind. Found a solution.

Reply:
So apparently axios serializes by default to JSON so that is why this is happening: https://stackoverflow.com/questions/59922431/axios-in-vue-js-send-request-as-string-not-object
As described here there are two options, use qs or just pass it as a PathParameter. I would suggest to go with the PathParameter as that is mostly used already in the Board and it would be consistent.
However I can be convinced that POST request should have a body, like described here:
https://medium.com/@LadyNoBug/api-design-f37bb37acf13
or here:
https://dzone.com/articles/rest-api-path-vs-request-body-parameters
But for one String this still seems like an overkill for me, the PathParameter would simplify the code by a lot.
What is your opinion?

Reply:
I made it to work by setting the header `Content-Type` to `text/plain` change the given controller method to consume these kind of values:

```
        var config = {
            headers: {
                'Content-Type': 'text/plain'
            }
        };
        axios.post(this.backendUrls.createRoom(), this.author, config)
```
        var config = {
            headers: {
                'Content-Type': 'text/plain'
            }
        };
        axios.post(this.backendUrls.createRoom(), this.author, config).then(...)
```
@RequestMapping(path = ROOM, method = RequestMethod.POST, consumes = MediaType.TEXT_PLAIN_VALUE)
    public ResponseEntity<RoomDetails> createRoom(@RequestBody String author) {}
```

I'd support to make POST requests have a body and keep the details there. Feels more natural for me, but also can modify it so it would look and work like at the BoardController.

Reply:
Ok, lets keep it in the body for POST requests, I will modify my endpoints in board as well like this.